timemachine/cpp/src/wrap_kernels.cpp

```diff
@@ -30,6 +30,35 @@
 
 namespace py = pybind11;
 
+// A utility to make sure that the coords and box shapes are correct
+void verify_coords_and_box(
+    const py::array_t<double, py::array::c_style> &coords, const py::array_t<double, py::array::c_style> &box) {
+    size_t coord_dimensions = coords.ndim();
+    if (coord_dimensions != 2) {
+        throw std::runtime_error("coords dimensions must be 2");
+    }
+    if (coords.shape(coord_dimensions - 1) != 3) {
+        throw std::runtime_error("coords must have a shape that is 3 dimensional");
+    }
+    if (box.ndim() != 2) {
+        throw std::runtime_error("box dimensions must be 2");
+    }
+
+    if (box.shape(0) != 3 || box.shape(1) != 3) {
+        throw std::runtime_error("box must be 3x3");
+    }
```

Comment:
nit: could these be combined like
```suggestion
    if (box.ndim() != 2 || box.shape(0) != 3 || box.shape(1) != 3) {
        throw std::runtime_error("box must be 3x3");
    }
```